### PR TITLE
[WIP] removes index.php from URL's for #5578

### PR DESCRIPTION
**Pull request by cdowdy**
[WIP] Removes index.php from urls. Nginx would need documentation done for bolt.cm/docs.

Fixes URL'ls like:  

https://site.com/index.php  -> https://site.com
https://site.com/index.php/page/zenonis-est-inquam-hoc-stoici -> https://site.com/page/zenonis-est-inquam-hoc-stoici

Fixes: #5578

Details
-------

I don't know, or think at least that the part dealing with http authorization would be needed by bolt. I don't think it does but I left it there because I don't know haha. Figured you all would be better suited to determine that. 

```
 RewriteCond %{HTTP:Authorization} .
 RewriteRule ^ - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]  
```

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -27,10 +27,36 @@ Options -Indexes
...
+  # to the start page because your Apache does not expose the REDIRECT_STATUS
+  # environment variable, you have 2 choices:
+  # - disable this feature by commenting the following 2 lines or
+  # - use Apache >= 2.3.9 and replace all L flags by END flags and remove the
+  #   following RewriteCond (best solution)
+  RewriteCond %{ENV:REDIRECT_STATUS} ^$
+  RewriteRule ^index\.php(?:/(.*)|$) %{ENV:BASE}/$1 [R=301,L]
+
+  # If the requested filename exists, simply serve it.
+  # We only want to let Apache serve files and not directories.
```
Comment (cdowdy) [inline .htaccess]: After thinking about this I'm not sure this is what you all want.  

Originally we had  

```  
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d  
```  

rewrite request filename that is not a directory and not a file. This rewrite serves files and NOT directories

The comments from symfony say  

> We only want to let Apache serve files and not directories.  

thats contradictory to what was there before no? maybe I've talked myself into a circle I dunno.

**Comment by CarsonF** [comment]
I could go either way on this. And really this only affects new installs (I don't count git installs as they should only be for dev) so this should go in the `bolt/composer-install` repo.

File: .htaccess (hunk 1 above)
Comment (SahAssar) [inline .htaccess]: I can't think of a situation we'd want to serve directories, or is there some point I'm missing?

**Comment by GawainLynch** [comment]
So I happened to end up on #1406 just now … It got me thinking about approach broadly.

We now base all our supported installs from a Composer project. This repository is really now the library that drives (by default of course) that project… ergo, I think @CarsonF is 100% correct and this should be in the [`bolt/composer-install`](https://github.com/bolt/composer-install) repository.

Proposed plan of action:
  1. Merge this in here, contributor credit where it's due … @cdowdy has put a lot of research work in here
  2. Move file from `bolt/bolt` to `bolt/composer-install`
    1. PR to remove from `bolt/bolt` and replace with absolute base minimum (enough for CI testing and core development)
    2. PR to add collective `.htaccess` to `bolt/composer-install`
  3. Update any documentation that needs it
  4. Profit!

**Comment by CarsonF** [comment]
Yeah I mean really in the long term goal was/is to restructure bolt/bolt to match the composer-install repo (with a couple extra folders of course).

**Comment by bobdenotter** [comment]
I agree with Gawain's approach. Seems like the quickest improvement for right now.

**Comment by CarsonF** [comment]
It could just be committed directly to `composer-install` with @cdowdy as the author. GitHub will give him credit for that.

**Comment by GawainLynch** [comment]
Credit in the main repo … but apart from that, the PR isn't a whole file, so that'd get fun quickly, no?

**Comment by CarsonF** [comment]
https://github.com/bolt/bolt/pull/5605.diff
https://github.com/bolt/bolt/pull/5605.patch

**Comment by GawainLynch** [comment]
@bobdenotter & @CarsonF can one of you please PR what need as a minimum `.htaccess` file in root to get a developer up and going please :beers:

**Comment by GawainLynch** [comment]
Oh, targeting `release/3.2`

**Pull request merged**